pymc/gp/cov.py

```diff
@@ -179,13 +179,13 @@ def n_dims(self) -> int:
         """The dimensionality of the input, as taken from the
         `active_dims`.
         """
-        # Evaluate lazily in-case this changes.
+        # Evaluate lazily in case this changes.
         return len(self.active_dims)
 
     def _slice(self, X, Xs=None):
         xdims = X.shape[-1]
         if isinstance(xdims, Variable):
-            [xdims] = constant_fold([xdims])
+            [xdims] = constant_fold([xdims], raise_not_constant=False)
         if self.input_dim != xdims:
```

Comment:
Your code probably works but this may be a bit more readable?

```suggestion
         # If xdims can't be fuly constant folded, it will be a TensorVariable
         if isinstance(xdims, int) and self.input_dim != xdims:
```